src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -158,6 +159,40 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
       const auto value2 = boost::get<T>(*variant_value2);
       return value >= max || value2 <= min;
     }
+    case PredicateCondition::Like: {  // NOLINTNEXTLINE - clang-tidy doesn't like the else path of the if constexpr
```

Comment:
Using this comment to block this PR for now. We face the issue that `USA%` is not pruned when ever just a single value with a lower case letter exists, because the default ASCII-based sorting places all upper case values before lower case values.

Any ideas? Shall we only store lower case (`lower(value)`) in the min/max filters?

@Alexander-Dubrawski : can you add a test case for that problem please?

Reply:
Do you really want to open Pandora's box of collected collations? ;)

Reply:
Thought about it a bit more - ignoring the case for statistics might generally make sense. The simple approach would break the characters between Z and a though.

Reply:
Do you have an example in mind where that would break?

Reply:
* Take a segment that only contains `"["`. Thus, both min and max are `"["`.
* `does_not_contain(Between, "A", "c")` would currently check for `max < "A" or min >= "d"`. This is false, so we cannot guarantee that the search value is not contained.
* If we normalize everything to lower case, it would check for `max < "a" or min >= "d"`. This is true, but shouldn't be.

I can't come up with something that breaks with (Not)Like - at the same time I cannot prove that it does *not* break, so I don't have a good feeling about that either.

I wonder if anyone ever wrote something about this before.

Reply:
> * If we normalize everything to lower case, it would check for `max < "a" or min >= "d"`. This is true, but shouldn't be.

Why shouldn't it be true?

Reply:
Because if we return `true`, we promise that no value between `A` and `c` is contained. That is false, because `[` is between `A` and `c`.

Reply:
Got it. Hmm ... never thought about betweens. That's ugly.

Not happy about using this for LIKEs either unless we know for sure. I did a short search for something in the literature, but could not find anything.

Reply:
#1536 \*cough* ;)

Reply:
Well that another topic (distance between strings), or do you mean something else?

Reply:
If we had RangeFilters for strings, this wouldn't be an issue.

Reply:
Sure? Wouldn’t it the same times n?

Reply:
I'd expect to end up, e.g., with a range containing the words starting with upper case and another one with the lower case ones.

Reply:
Well ok, one could enforce that. But the range between `Z` and `a` is pretty small.
 Good chance that this would end up as one range.

Reply:
We have discussed this topic yesterday. We think the current solution with using ASCII-based sorting is sufficient for now. While we could improve it further, it would be a lot of work and JOB already profits from the current implementation (I'll post the final benchmarks around noon).

@Alexander-Dubrawski: can you add a comment to the like pruning about that issue and to the most recent test, which is then expecting a non-prunable predicate?

Reply:
Ok 👍